packages/syntax-core/src/TapArea/TapArea.test.tsx

```diff
@@ -0,0 +1,170 @@
+import { screen, render } from "@testing-library/react";
+import TapArea from "./TapArea";
+import { expect, vi } from "vitest";
+import userEvent from "@testing-library/user-event";
+import { createRef } from "react";
+
+describe("tapArea", () => {
+  it("renders successfully", () => {
+    const { baseElement } = render(
+      <TapArea
+        onClick={() => {
+          /* empty */
+        }}
+      />,
+    );
+    expect(baseElement).toBeTruthy();
+  });
+
+  it("renders an role=button element", async () => {
+    render(
+      <TapArea
+        onClick={() => {
+          /* empty */
+        }}
+      />,
+    );
+    const tapArea = await screen.findAllByRole("button");
+    expect(tapArea).toHaveLength(1);
+  });
+
+  it("renders its children", async () => {
+    render(
+      <TapArea
+        onClick={() => {
+          /* empty */
+        }}
+      >
+        <div data-testid="tap-area-child-testid">Continue</div>
+      </TapArea>,
+    );
+    const child = await screen.findByTestId("tap-area-child-testid");
+    expect(child).toHaveTextContent("Continue");
+  });
+
+  it("correctly applies fullWidth when set", async () => {
+    render(
+      <TapArea
+        data-testid="tap-area-testid"
+        onClick={() => {
+          /* empty */
+        }}
```

Comment:
aside, non-blocking, side-discussion:

Seeing all the empty `onClick` handlers in the test here makes we wonder if we shouldn't make `onClick` an optional prop.  `onClick` is always optional for the HTMLElements that implements it.